### PR TITLE
Implementa avaliação fisioterapêutica do paciente

### LINKED ISSUE #50: Implementar Avaliação Fisioterapêutica do paciente
## Descrição do problema ou necessidade

O sistema precisa registrar a avaliação fisioterapêutica do paciente, contendo informações técnicas coletadas pelo fisioterapeuta. Essa avaliação serve como base para diagnóstico fisioterapêutico, definição de objetivos, plano de tratamento e acompanhamento da evolução clínica.

## Objetivo da atividade

Implementar o módulo de Avaliação Fisioterapêutica vinculado ao paciente, permitindo registrar, consultar e atualizar dados técnicos da avaliação inicial ou periódica.

## Escopo da implementação

- Criar entidade/modelo de Avaliação Fisioterapêutica relacionada ao Paciente.
- Criar DTOs de request e response.
- Criar repository, service e controller REST.
- Implementar endpoints para:
  - Criar avaliação para um paciente.
  - Buscar avaliação por ID.
  - Listar avaliações de um paciente.
  - Atualizar avaliação.
- Definir campos mínimos:
  - Data da avaliação.
  - Queixa funcional.
  - Avaliação postural.
  - Mobilidade articular.
  - Força muscular.
  - Flexibilidade.
  - Equilíbrio.
  - Coordenação motora.
  - Padrão respiratório.
  - Escala de dor.
  - Testes funcionais realizados.
  - Diagnóstico fisioterapêutico.
  - Observações gerais.
- Validar existência do paciente antes de criar a avaliação.
- Permitir múltiplas avaliações por paciente para manter histórico clínico.

## Critérios de aceite

- Deve ser possível criar uma avaliação fisioterapêutica para um paciente existente.
- Não deve ser possível criar avaliação para paciente inexistente.
- Deve ser possível listar todas as avaliações de um paciente.
- Deve ser possível consultar uma avaliação específica pelo ID.
- Deve ser possível atualizar os dados de uma avaliação.
- A avaliação deve armazenar a data em que foi realizada.
- O sistema deve validar campos obrigatórios.
- Os endpoints devem seguir o padrão REST do projeto.

## Sugestão de validação/testes

- Teste unitário para criação de avaliação com paciente válido.
- Teste para erro ao criar avaliação com paciente inexistente.
- Teste para listagem de avaliações por paciente.
- Teste para atualização de avaliação.
- Teste para validação de escala de dor, caso exista regra de 0 a 10.
- Teste de integração do controller, se aplicável.

## Observações técnicas

- Avaliar uso de campos textuais para dados subjetivos e campos numéricos para escalas mensuráveis.
- Escala de dor pode seguir padrão de 0 a 10.
- A avaliação deve ser independente da anamnese, mas ambas devem estar vinculadas ao paciente.
- Considerar que futuramente o módulo poderá alimentar relatórios de evolução.

## Labels sugeridas

- `enhancement`
- `backend`
- `avaliacao-fisioterapeutica`
- `prontuario`

### PR DESCRIPTION
**Pull request by FabioCarlesso**
## Resumo da alteração

- Adiciona o módulo de Avaliação Fisioterapêutica com entidade JPA, DTOs, repository, service e controller REST.
- Cria endpoints para cadastrar, buscar por ID, listar por paciente e atualizar avaliações.
- Adiciona migration Flyway para a tabela `avaliacoes_fisioterapeuticas`, incluindo índice por `paciente_id` e validação de `escala_dor` entre 0 e 10.
- Atualiza README e `docs/context.md` com endpoints, modelo de dados, regras e testes.

## Motivo da mudança

Permitir registrar avaliações fisioterapêuticas técnicas vinculadas ao paciente, mantendo histórico clínico com múltiplas avaliações por paciente, conforme solicitado na issue #50.

## Testes executados

- `JAVA_HOME=~/jdk mvn -Dtest=AvaliacaoFisioterapeuticaServiceTest,AvaliacaoFisioterapeuticaControllerTest test`
- `JAVA_HOME=~/jdk mvn test` (261 testes, 0 falhas)
- `JAVA_HOME=~/jdk mvn -DskipTests package`

## Arquivos principais alterados

- `src/main/java/com/carlesso/pilatesapi/entity/AvaliacaoFisioterapeutica.java`
- `src/main/java/com/carlesso/pilatesapi/controller/AvaliacaoFisioterapeuticaController.java`
- `src/main/java/com/carlesso/pilatesapi/service/AvaliacaoFisioterapeuticaService.java`
- `src/main/java/com/carlesso/pilatesapi/dto/AvaliacaoFisioterapeutica*.java`
- `src/main/java/com/carlesso/pilatesapi/repository/AvaliacaoFisioterapeuticaRepository.java`
- `src/main/resources/db/migration/V15__create_avaliacoes_fisioterapeuticas_table.sql`
- `src/test/java/com/carlesso/pilatesapi/service/AvaliacaoFisioterapeuticaServiceTest.java`
- `src/test/java/com/carlesso/pilatesapi/controller/AvaliacaoFisioterapeuticaControllerTest.java`
- `README.md`
- `docs/context.md`

## Referência

Closes #50

### CODE REVIEW

**Review by FabioCarlesso — COMMENTED** [review]
## Code Review — PR #57: Avaliação Fisioterapêutica

Overall this is a well-structured implementation that follows the existing project conventions closely. The layered architecture (controller → service → repository → entity), Flyway migration, DTO pattern, and test coverage all look solid.

That said, I found **2 high severity** and **several medium/low severity** issues across performance, bug potential, and Spring Boot practices that should be addressed before merging. See inline comments for details.

File: src/main/java/com/carlesso/pilatesapi/service/AvaliacaoFisioterapeuticaService.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    private AvaliacaoFisioterapeutica encontrar(Long id) {
+        return avaliacaoRepository.findByIdAndPacienteAtivoTrue(id)
+                .orElseThrow(() -> new ResourceNotFoundException("Avaliação fisioterapêutica não encontrada: " + id));
+    }
+
+    private void validarCampoObrigatorio(String campo, String valor) {
+        if (valor != null && valor.isBlank()) {
+            throw new IllegalArgumentException(campo + " não pode ser vazio");
```
Comment (FabioCarlesso) [inline src/main/java/com/carlesso/pilatesapi/service/AvaliacaoFisioterapeuticaService.java]: **🔴 Bug — `IllegalArgumentException` may produce HTTP 500 (High)**

- **Problem:** `validarCampoObrigatorio` throws `IllegalArgumentException` when a mandatory field is blank. Unless `GlobalExceptionHandler` explicitly maps this exception to HTTP 400, Spring will treat it as an unhandled server error and return 500.
- **Why it matters:** The client sending a blank `queixaFuncional` would receive a 500 instead of the expected 400, breaking the API contract documented in the README.
- **Suggested improvement:** Either verify that `GlobalExceptionHandler` handles `IllegalArgumentException` → 400, or — since `AvaliacaoFisioterapeuticaUpdateDTO` already carries `@Pattern(regexp = ".*\\S.*")` annotations validated by `@Valid` at the controller level — remove the redundant `validarCampoObrigatorio` calls from the service entirely. Dual validation at both controller and service layers creates inconsistency.

File: src/main/java/com/carlesso/pilatesapi/service/AvaliacaoFisioterapeuticaService.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        if (dto.equilibrio() != null) avaliacao.setEquilibrio(dto.equilibrio());
+        if (dto.coordenacaoMotora() != null) avaliacao.setCoordenacaoMotora(dto.coordenacaoMotora());
+        if (dto.padraoRespiratorio() != null) avaliacao.setPadraoRespiratorio(dto.padraoRespiratorio());
+        if (dto.escalaDor() != null) avaliacao.setEscalaDor(dto.escalaDor());
+        if (dto.testesFuncionaisRealizados() != null) avaliacao.setTestesFuncionaisRealizados(dto.testesFuncionaisRealizados());
+        if (dto.diagnosticoFisioterapeutico() != null) avaliacao.setDiagnosticoFisioterapeutico(dto.diagnosticoFisioterapeutico());
+        if (dto.observacoesGerais() != null) avaliacao.setObservacoesGerais(dto.observacoesGerais());
+        avaliacao.setDataAtualizacao(LocalDateTime.now());
+
+        return AvaliacaoFisioterapeuticaResponseDTO.from(avaliacao);
```
Comment (FabioCarlesso) [inline src/main/java/com/carlesso/pilatesapi/service/AvaliacaoFisioterapeuticaService.java]: **🟡 Spring Boot Best Practice — Implicit dirty checking without `save()` (Medium)**

- **Problem:** `atualizar` modifies the managed entity and returns without calling `avaliacaoRepository.save(avaliacao)`. The update is persisted via JPA dirty checking when the `@Transactional` method commits.
- **Why it matters:** This is a subtle implicit side-effect. If this method is ever refactored to use `propagation = REQUIRES_NEW` or the transaction is detached, changes will silently be lost. It also diverges from the `criar` method which calls `save()` explicitly, making the codebase inconsistent.
- **Suggested improvement:** Add an explicit `return AvaliacaoFisioterapeuticaResponseDTO.from(avaliacaoRepository.save(avaliacao));` to keep the pattern consistent with `criar` and signal intent clearly.

File: src/main/java/com/carlesso/pilatesapi/service/AvaliacaoFisioterapeuticaService.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    @Transactional(readOnly = true)
+    public AvaliacaoFisioterapeuticaResponseDTO buscarPorId(Long id) {
+        return AvaliacaoFisioterapeuticaResponseDTO.from(encontrar(id));
+    }
+
+    @Transactional(readOnly = true)
+    public List<AvaliacaoFisioterapeuticaResponseDTO> listarPorPaciente(Long pacienteId) {
+        if (!pacienteRepository.existsByIdAndAtivoTrue(pacienteId)) {
```
Comment (FabioCarlesso) [inline src/main/java/com/carlesso/pilatesapi/service/AvaliacaoFisioterapeuticaService.java:61]: **🟡 Performance — Two queries where one suffices (Medium)**

- **Problem:** `listarPorPaciente` first calls `existsByIdAndAtivoTrue(pacienteId)` (query 1), then `findByPaciente...` (query 2). If the patient exists but has no evaluations, both queries run and the second returns an empty list.
- **Why it matters:** This is an unnecessary round-trip to the database on every request.
- **Suggested improvement:** Drop the existence check and rely solely on the list query. If you still want to distinguish 'patient not found' from 'patient has no evaluations', keep the check — but document the trade-off. Alternatively, combined with the JOIN FETCH suggestion above, a single query can handle both concerns.

File: src/main/java/com/carlesso/pilatesapi/dto/AvaliacaoFisioterapeuticaUpdateDTO.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package com.carlesso.pilatesapi.dto;
+
+import jakarta.validation.constraints.Max;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.Pattern;
+import java.time.LocalDate;
+
+public record AvaliacaoFisioterapeuticaUpdateDTO(
+        LocalDate dataAvaliacao,
+        @Pattern(regexp = ".*\\S.*", message = "queixaFuncional não pode ser vazio")
```
Comment (FabioCarlesso) [inline src/main/java/com/carlesso/pilatesapi/dto/AvaliacaoFisioterapeuticaUpdateDTO.java:10]: **🟡 Code Quality — Duplicate validation (Medium)**

- **Problem:** `@Pattern(regexp = ".*\\S.*")` here rejects blank strings at controller level via `@Valid`, and `validarCampoObrigatorio` in the service does the same check again.
- **Why it matters:** The two mechanisms are not synchronized — if one is changed independently, the behavior diverges. It also causes confusion about which layer owns the validation.
- **Suggested improvement:** Choose one layer. Given that this is a format constraint (not a business rule), the DTO `@Pattern` at the controller level is the right place. Remove `validarCampoObrigatorio` from the service.

File: src/main/java/com/carlesso/pilatesapi/entity/AvaliacaoFisioterapeutica.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    public void setEquilibrio(String equilibrio) { this.equilibrio = equilibrio; }
+
+    public String getCoordenacaoMotora() { return coordenacaoMotora; }
+    public void setCoordenacaoMotora(String coordenacaoMotora) { this.coordenacaoMotora = coordenacaoMotora; }
+
+    public String getPadraoRespiratorio() { return padraoRespiratorio; }
+    public void setPadraoRespiratorio(String padraoRespiratorio) { this.padraoRespiratorio = padraoRespiratorio; }
+
+    public Integer getEscalaDor() { return escalaDor; }
+    public void setEscalaDor(Integer escalaDor) { this.escalaDor = escalaDor; }
```
Comment (FabioCarlesso) [inline src/main/java/com/carlesso/pilatesapi/entity/AvaliacaoFisioterapeutica.java:103]: **🟢 Spring Boot Best Practice — Use `@PrePersist`/`@PreUpdate` for audit fields (Low)**

- **Problem:** `dataCriacao` and `dataAtualizacao` are set manually in the service (`avaliacao.setDataCriacao(LocalDateTime.now())`). If someone creates or updates an entity through any other code path (e.g. a test, migration, future batch job), these fields won't be set automatically.
- **Why it matters:** Audit timestamps are an entity concern, not a service concern. Putting them in the service creates an invisible contract that every caller must remember.
- **Suggested improvement:** Add lifecycle callbacks on the entity:
```java
@PrePersist
void prePersist() { this.dataCriacao = LocalDateTime.now(); }

@PreUpdate
void preUpdate() { this.dataAtualizacao = LocalDateTime.now(); }
```
This is consistent with how Hibernate-managed timestamps should work.

File: src/main/java/com/carlesso/pilatesapi/repository/AvaliacaoFisioterapeuticaRepository.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface AvaliacaoFisioterapeuticaRepository extends JpaRepository<AvaliacaoFisioterapeutica, Long> {
+
+    Optional<AvaliacaoFisioterapeutica> findByIdAndPacienteAtivoTrue(Long id);
+
+    List<AvaliacaoFisioterapeutica> findByPacienteIdAndPacienteAtivoTrueOrderByDataAvaliacaoDescIdDesc(Long pacienteId);
```
Comment (FabioCarlesso) [inline src/main/java/com/carlesso/pilatesapi/repository/AvaliacaoFisioterapeuticaRepository.java]: **🟢 Code Quality — Derived query name is excessively long (Low)**

- **Problem:** `findByPacienteIdAndPacienteAtivoTrueOrderByDataAvaliacaoDescIdDesc` is 72 characters. Spring Data supports this, but it is hard to read, refactor, and IDE-navigate.
- **Why it matters:** Maintainability. The method name encodes the entire query logic, making it brittle if column names change.
- **Suggested improvement:** Replace with a `@Query` annotation and a shorter, descriptive method name:
```java
@Query("SELECT a FROM AvaliacaoFisioterapeutica a JOIN FETCH a.paciente p "
     + "WHERE p.id = :pacienteId AND p.ativo = true "
     + "ORDER BY a.dataAvaliacao DESC, a.id DESC")
List<AvaliacaoFisioterapeutica> findAtivasByPacienteOrdenadas(@Param("pacienteId") Long pacienteId);
```
This also solves the N+1 query issue described in the service comment above.

File: src/test/java/com/carlesso/pilatesapi/service/AvaliacaoFisioterapeuticaServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+        p.setNome("Maria Souza");
+        p.setEmail("maria@email.com");
+        p.setCpf("12345678900");
+        setId(p, 1L);
+        return p;
+    }
+
+    private AvaliacaoFisioterapeutica avaliacao(Paciente paciente) {
+        AvaliacaoFisioterapeutica a = new AvaliacaoFisioterapeutica();
+        a.setPaciente(paciente);
```
Comment (FabioCarlesso) [inline src/test/java/com/carlesso/pilatesapi/service/AvaliacaoFisioterapeuticaServiceTest.java:51]: **🟡 Tests — Missing edge-case coverage (Medium)**

The service test suite has 8 cases. The following scenarios are missing:

1. **`atualizar` with non-existent ID** — should throw `ResourceNotFoundException`. Currently only `buscarPorId` has a 'not found' negative test; `atualizar` does not.
2. **`atualizar` with blank `diagnosticoFisioterapeutico`** — only the blank `queixaFuncional` path is tested.
3. **`listarPorPaciente` returns empty list** — a patient with no evaluations should return `[]`, not an error. This happy-but-empty path is not tested.

These are the exact scenarios most likely to regress silently because they exercise the boundary between 'no data' and 'error' states.

File: src/test/java/com/carlesso/pilatesapi/controller/AvaliacaoFisioterapeuticaControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+import com.carlesso.pilatesapi.exception.ResourceNotFoundException;
+import com.carlesso.pilatesapi.service.AvaliacaoFisioterapeuticaService;
+import com.carlesso.pilatesapi.service.CustomUserDetailsService;
+import com.carlesso.pilatesapi.service.JwtService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
```
Comment (FabioCarlesso) [inline src/test/java/com/carlesso/pilatesapi/controller/AvaliacaoFisioterapeuticaControllerTest.java:15]: **🟢 Tests — Security filters disabled (Low)**

- **Problem:** `@AutoConfigureMockMvc(addFilters = false)` disables the JWT filter for all controller tests. This is consistent with other controller tests in the project.
- **Why it matters:** It means zero controller-level tests verify that unauthenticated requests are rejected. An accidental removal of the security config on this route would not be caught.
- **Suggested improvement:** This is a project-wide gap, not specific to this PR. Consider adding at least one security integration test (like `SecurityIntegrationTest`) that verifies `/avaliacoes-fisioterapeuticas` routes return 401 without a valid token.

File: src/main/java/com/carlesso/pilatesapi/controller/AvaliacaoFisioterapeuticaController.java [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    public ResponseEntity<AvaliacaoFisioterapeuticaResponseDTO> buscarPorId(
+            @Parameter(description = "ID da avaliação fisioterapêutica", required = true) @PathVariable Long id) {
+        return ResponseEntity.ok(service.buscarPorId(id));
+    }
+
+    @Operation(summary = "Listar avaliações fisioterapêuticas por paciente")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "Avaliações encontradas"),
+            @ApiResponse(responseCode = "404", description = "Paciente não encontrado")
+    })
```
Comment (FabioCarlesso) [inline src/main/java/com/carlesso/pilatesapi/controller/AvaliacaoFisioterapeuticaController.java:63]: **🟢 Performance — No pagination on listing endpoint (Low)**

- **Problem:** `listarPorPaciente` returns `List<AvaliacaoFisioterapeuticaResponseDTO>` with no size limit. A patient under long-term treatment could accumulate dozens or hundreds of evaluations over time.
- **Why it matters:** Uncapped list endpoints are a common source of memory pressure and slow responses as data grows.
- **Suggested improvement:** Consider accepting a `Pageable` parameter and returning `Page<AvaliacaoFisioterapeuticaResponseDTO>` using Spring Data's standard pagination. Low priority for now but worth tracking as data grows.
